diff --git a/src/Plugin/ContextReaction/Blocks.php b/src/Plugin/ContextReaction/Blocks.php
index 1f72a02..905466b 100644
--- a/src/Plugin/ContextReaction/Blocks.php
+++ b/src/Plugin/ContextReaction/Blocks.php
@@ -219,7 +219,7 @@ class Blocks extends ContextReactionPluginBase implements ContainerFactoryPlugin
           'route_parameters' => [
             'context' => $configuration['context_id'],
             'reaction_id' => 'blocks',
-            'block_id' => $block->getConfiguration()['uuid'],
+            'block_id' => $block->getConfiguration()['uuid'] ?? $this->uuid->generate(),
           ],
         ];
 
